src/query/api/v1/handler/prometheus/remote/read_test.go

```diff
@@ -71,8 +71,10 @@ func setupServer(t *testing.T) *httptest.Server {
 
 func readHandler(store storage.Storage, timeoutOpts *prometheus.TimeoutOpts) *PromReadHandler {
 	opts := handler.FetchOptionsBuilderOptions{Limit: 100}
+	engineOpts := executor.NewEngineOpts().SetStore(store).SetCostScope(tally.NewTestScope("test", nil)).
```

Comment:
nit: may be easier to pull engine generation into a new function

(or better yet delete it and use a mock here instead?)

Reply:
Let me pull into a func for now and i can add mocks in another pr.